### PR TITLE
Create helper methods and begin refactor 

### PR DESCRIPTION
**Pull request by tyladevon**

### CODE REVIEW

File: lib/league_statistics.rb [hunk 1]
```diff
@@ -1,44 +1,50 @@
...
+  end
+
+  def goal_avg_helper
+    games_by_team = @game_teams.group_by(&:team_id)
     goal_avg_by_team = {}
     games_by_team.each do |team_id, games_arr|
-      total_goals = games_arr.sum do |game|
-        game.goals
-      end
+      total_goals = games_arr.sum(&:goals)
```
Comment (lrs8810) [inline lib/league_statistics.rb:17]: This looks awesome!

File: lib/team_statistics.rb [hunk 2]
```diff
@@ -240,21 +158,12 @@ def head_to_head(team_id)
   end
 
   def seasonal_summary(team_id)
-    all_seasons = @games.inject([]) do |seasons, (game_id, game)|
+    all_seasons = @games.inject([]) do |seasons, (_, game)|
```
Comment (lrs8810) [inline lib/team_statistics.rb:161]: Nice implementation!

**Review by lrs8810 — APPROVED** [review]

File: lib/league_statistics.rb [hunk 3]
```diff
@@ -64,116 +70,43 @@ def best_defense
...
-    end
-
-    teams = home_teams.merge(away_teams) { |team_id, home_team_val, away_team_val| home_team_val + away_team_val}
-
-    goal_avg_by_team = {}
-    teams.each do |team_id, goals_arr|
-      total_goals = goals_arr.sum
-      goal_avg_by_team[team_id] = (total_goals.to_f / goals_arr.length).round(2)
-    end
-
```
Comment (grwthomps) [inline lib/league_statistics.rb:100]: 25 lines of code removed! Woohoo!

File: lib/league_statistics.rb [hunk 4]
```diff
@@ -64,116 +70,43 @@ def best_defense
...
-        away_games_goals = games_arr.find_all do |game|
-          game.hoa == "away"
-        end.sum { |game| game.goals }
-        goal_avg_by_team[team_id] = (away_games_goals.to_f / games_arr.length).round(2)
-      end
-
-      highest_team_id = goal_avg_by_team.key(goal_avg_by_team.values.min_by { |value| value})
-      @teams.select do |team_id, team|
-        highest_team_id == team_id
-      end[highest_team_id].team_name
```
Comment (grwthomps) [inline lib/league_statistics.rb:158]: Huge refactor gains! Great job!

File: lib/team_statistics.rb [hunk 5]
```diff
@@ -39,16 +41,7 @@ def best_season(team_id)
...
-    filtered_games = []
-    @games.each do |game_id, game|
-      if game.home_team_id == team_id || game.away_team_id == team_id
-        filtered_games.push(game)
-      end
-    end
-
-    games_by_season = filtered_games.group_by do |game|
-      game.season
-    end
```
Comment (grwthomps) [inline lib/team_statistics.rb:51]: Awesome refactor!

**Review by grwthomps — APPROVED** [review]

**Pull request merged**